File: payola/web/server/public/javascripts/runtime/s2js/runtime/scala/collection/immutable/StringOps.js
goog.provide('scala.collection.immutable.StringOps');
goog.require('scala.collection.Seq');
goog.require('scala.collection.SeqCompanion');
goog.require('scala.collection.immutable.List');
scala.collection.immutable.StringOps = function(x) {
var self = this;
self.x = x;
goog.base(self);
self.initializeInternalJsArray(self.x);
};
goog.inherits(scala.collection.immutable.StringOps, scala.collection.Seq);
scala.collection.immutable.StringOps.prototype.newInstance = function() {
var self = this;
return scala.collection.immutable.StringOps.empty();
};
scala.collection.immutable.StringOps.prototype.initializeInternalJsArray = function(value) {
var self = this;
self.setInternalJsArray(value.split(''))};
scala.collection.immutable.StringOps.prototype.repr = function() {
var self = this;
return self.getInternalJsArray().join();};
scala.collection.immutable.StringOps.prototype.toBoolean = function() {
var self = this;
return self.x == 'true';};
scala.collection.immutable.StringOps.prototype.toByte = function() {
var self = this;
return parseInt(self.x);};
scala.collection.immutable.StringOps.prototype.toShort = function() {
var self = this;
return parseInt(self.x);};
scala.collection.immutable.StringOps.prototype.toInt = function() {
var self = this;
return parseInt(self.x);};
scala.collection.immutable.StringOps.prototype.toLong = function() {
var self = this;
return parseInt(self.x);};
scala.collection.immutable.StringOps.prototype.toFloat = function() {
var self = this;
return parseFloat(self.x);};
scala.collection.immutable.StringOps.prototype.toDouble = function() {
var self = this;
return parseFloat(self.x);};
scala.collection.immutable.StringOps.prototype.toString = function() {
var self = this;
return self.x;
};
scala.collection.immutable.StringOps.prototype.__class__ = new s2js.Class('scala.collection.immutable.StringOps', [scala.collection.Seq]);
goog.object.extend(scala.collection.immutable.StringOps, new scala.collection.SeqCompanion());
scala.collection.immutable.StringOps.empty = function() {
var self = this;
return new scala.collection.immutable.StringOps('');
};
scala.collection.immutable.StringOps.$apply = function() {
var self = this;
var xs = scala.collection.immutable.List.fromJsArray([].splice.call(arguments, 0, arguments.length - 0));
return self.fromJsArray(xs.getInternalJsArray());};
scala.collection.immutable.StringOps.__class__ = new s2js.Class('scala.collection.immutable.StringOps', [scala.collection.SeqCompanion]);

File: payola/s2js/runtime/client/src/main/scala/s2js/runtime/client/scala/collection/immutable/StringOps.scala
package s2js.runtime.client.scala.collection.immutable

import s2js.compiler.javascript

object StringOps extends s2js.runtime.client.scala.collection.SeqCompanion[StringOps]
{
    def empty = new StringOps("")

    @javascript("return self.fromJsArray(xs.getInternalJsArray());")
    def apply(xs: Any*): Any = null
}

class StringOps(val str: java.lang.String) extends s2js.runtime.client.scala.collection.Seq
{
    val x = Option(str).getOrElse("")

    initializeInternalJsArray(x)

    def newInstance = StringOps.empty

    @javascript("""self.setInternalJsArray(value.split(''))""")
    def initializeInternalJsArray(value: String) {}

    @javascript("return self.getInternalJsArray().join('');")
    def repr: String = ""

    @javascript("return self.x == 'true';")
    def toBoolean: Boolean = false

    @javascript("return parseInt(self.x);")
    def toByte: Byte = 0

    @javascript("return parseInt(self.x);")
    def toShort: Short = 0

    @javascript("return parseInt(self.x);")
    def toInt: Int = 0

    @javascript("return parseInt(self.x);")
    def toLong: Long = 0

    @javascript("return parseFloat(self.x);")
    def toFloat: Float = 0

    @javascript("return parseFloat(self.x);")
    def toDouble: Double = 0.0

    @javascript("return self.repr().split(pattern);")
    def split(pattern: String): Any = null

    @javascript("return self.x.replace(pattern, replacement);")
    def replaceAllLiterally(pattern: String, replacement: String) = null

    @javascript(
        """
         var copiedArgs = [];
          for(var i=0; i<arguments.length; i++){
              copiedArgs.push(arguments[i]);
          }
         try {
             return vsprintf(self.str, copiedArgs);
         }catch(err){
            console.error("Error evaluating sprinf on '%s'", self.str);
            return self.str;
         }
        """
    )
    def format(args: Array[Any]) = ""

    override def toString = repr
}
